src/write.rs

```diff
@@ -548,9 +611,61 @@ where
                     }
                 }
             }
+            Operation::Retain(ref key, ref predicate) => {
+                if let Some(e) = inner.data.get_mut(key) {
+                    let mut del = 0;
+
```

Comment:
There are an awful lot of empty lines here that I don't think `rustfmt` would add automatically. I generally prefer to use empty lines to denote groups of logically connected statements, as opposed to using an extra newline between every line.